anchor/programs/bless-token/src/context/fund.rs

```diff
@@ -0,0 +1,186 @@
+        constraint = foundation.key() == bless_state.foundation,
+    )]
+    pub foundation: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == market_making.mint,
+        constraint = market_making.key() == bless_state.market_making,
+    )]
+    pub market_making: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == tgt_marketing.mint,
+        constraint = tgt_marketing.key() == bless_state.tgt_marketing,
+    )]
+    pub tgt_marketing: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == community_airdrop.mint,
+        constraint = community_airdrop.key() == bless_state.community_airdrop,
+    )]
+    pub community_airdrop: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == community_incentives.mint,
+        constraint = community_incentives.key() == bless_state.community_incentives,
+    )]
+    pub community_incentives: Box<Account<'info, TokenAccount>>,
+
+    pub system_program: Program<'info, System>,
+
+    pub token_program: Program<'info, Token>,
+}
+
+impl<'info> FundBlessToken<'info> {
+    pub fn transfer(&self, to: AccountInfo<'info>, amount: u64) -> Result<()> {
+        let cpi_ctx = CpiContext::new(
+            self.bless_mint.to_account_info(),
+            Transfer {
+                from: self.vault_account.to_account_info(),
+                to: to,
+                authority: self.bless_state.to_account_info(),
+            },
+        );
+
+        token::transfer(cpi_ctx, amount)?;
+
+        Ok(())
+    }
+
+    /// fund genesis token to packet accounts, for the rule only fund once.
+    pub fn fund_genesis(&mut self) -> Result<()> {
+        self.transfer(self.market_making.to_account_info(), MARKET_MAKING_FUND_FEE)?;
+        self.bless_state.market_making_flag = true;
+
+        self.transfer(self.tgt_marketing.to_account_info(), TGE_MARKETING_FUND_FEE)?;
+        self.bless_state.tgt_marketing_flag = true;
+
+        self.transfer(
+            self.community_airdrop.to_account_info(),
+            COMMUNITY_AIRDROP_FUND_FEE,
+        )?;
+        self.bless_state.community_airdrop_flag = true;
+
+        self.transfer(
+            self.community_incentives.to_account_info(),
+            COMMUNITY_INCENTIVES_FUN_FEE,
+        )?;
+        self.bless_state.community_incentives_flag = true;
+        self.bless_state.current_month = 1;
+        Ok(())
+    }
+
+    /// fund period, which setup the rule when initial.
+    pub fn fund_by_rule(&mut self) -> Result<()> {
+        let idx = self.bless_state.current_month as usize;
+        macro_rules! fund {
+            ($ro: ident, $a: ident, $fee: ident) => {
+                if self.bless_state.$ro.outcome.get(idx)? {
+                    self.transfer(self.$a.to_account_info(), PRE_SEED_SALE_FUND_FEE)?;
```

Comment:
shouldn't the `$fee` parameter be used?
```suggestion
                    self.transfer(self.$a.to_account_info(), $fee)?;
```

Reply:
In the first version design will be hard code to distribute to the account by rules. 
the v2 version we don't  need distribute to by contract ( will push soon)